pyanaconda/ui/gui/spokes/root_password.py

```diff
@@ -179,24 +179,26 @@ def mandatory(self):
         return not self._users_module.CheckAdminUserExists()
 
     def apply(self):
-        pw = self.password
+        if self.root_enabled and self.password:
+            # Set the root password.
+            self._users_module.SetCryptedRootPassword(crypt_password(self.password))
 
-        enabled = self.root_enabled
-        self._users_module.SetRootAccountLocked(not enabled)
+            # Unlock the root account.
+            self._users_module.SetRootAccountLocked(False)
 
-        if enabled:
+        else:
+            # Reset the root password.
+            self._users_module.ClearRootPassword()
+
+            # Lock the root account.
+            self._users_module.SetRootAccountLocked(True)
```

Comment:
Do we want to Lock the root if the root account is enabled but password is not set? I'm just not sure if that is the correct behavior :).

Reply:
I agree that porting it 1:1 is correct.

Reply:
Thanks for explanation.